src/BlockNLPModels.jl

```diff
@@ -0,0 +1,213 @@
+module BlockNLPModels
+
+using NLPModels
+using LinearAlgebra
+using SparseArrays
+
+export AbstractBlockNLPModel
```

Comment:
I would be careful here, and limit the number of `export` (ideally we should have 0 functions exported). A common rule is to not export the abstract types.

Reply:
I think exporting ``AbstractBlockNLPModel`` here is unavoidable, because, eventually ``BlockNLPAlgorithms`` be written with an understanding that a model being input to that package is an ``AbstractBlockNLPModel``. Won't we need this abstract type at that time?